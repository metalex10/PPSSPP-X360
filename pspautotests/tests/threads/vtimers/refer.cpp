#include "shared.h"

extern "C" int main(int argc, char *argv[]) {
	SceUID vtimer = sceKernelCreateVTimer("test", NULL);
	SceKernelVTimerInfo info;
	info.size = sizeof(info);

	checkpointNext("Objects");
	checkpoint("  Normal: %08x", sceKernelReferVTimerStatus(vtimer, &info));
	checkpoint("  Twice: %08x", sceKernelReferVTimerStatus(vtimer, &info));
	checkpoint("  NULL: %08x", sceKernelReferVTimerStatus(0, &info));
	checkpoint("  Invalid: %08x", sceKernelReferVTimerStatus(0xDEADBEEF, &info));
	sceKernelDeleteVTimer(vtimer);
	checkpoint("  Deleted: %08x", sceKernelReferVTimerStatus(vtimer, &info));

	vtimer = sceKernelCreateVTimer("test", NULL);
	checkpointNext("Sizes");
	int sizes[] = {-1, 0, 1, 4, 8, 9, 32, 64, 70, 72, 800};
	for (size_t i = 0; i < ARRAY_SIZE(sizes); ++i) {
		memset(&info, 0xCC, sizeof(info));
		info.size = sizes[i];
		int result = sceKernelReferVTimerStatus(vtimer, &info);
		if (result == 0) {
			checkpoint("  size %d => %d, %08x", sizes[i], info.size, info.common);
		} else {
			checkpoint("  size %d: Failed (%08x)", sizes[i], result);
		}
	}

	sceKernelDeleteVTimer(vtimer);
	return 0;
}