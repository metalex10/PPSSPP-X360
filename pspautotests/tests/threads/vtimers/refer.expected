[x] Objects
[x]   Normal: 00000000
[x]   Twice: 00000000
[x]   NULL: 800201be
[x]   Invalid: 800201be
[x]   Deleted: 800201be

[x] Sizes
[x]   size -1 => 72, 00000000
[x]   size 0 => 0, cccccccc
[x]   size 1 => 72, cccccccc
[x]   size 4 => 72, cccccccc
[x]   size 8 => 72, cccccccc
[x]   size 9 => 72, cccccccc
[x]   size 32 => 72, cccccccc
[x]   size 64 => 72, cccccccc
[x]   size 70 => 72, cccc0000
[x]   size 72 => 72, 00000000
[x]   size 800 => 72, 00000000
